hibernate-reactive-panache-quickstart/src/main/java/org/acme/hibernate/orm/panache/FruitResource.java

```diff
@@ -0,0 +1,136 @@
+package org.acme.hibernate.orm.panache;
+
+import java.util.List;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.reactive.RestPath;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.quarkus.panache.common.Sort;
+import io.smallrye.mutiny.Uni;
+
+@Path("fruits")
+@ApplicationScoped
+@Produces("application/json")
+@Consumes("application/json")
+public class FruitResource {
+
+    private static final Logger LOGGER = Logger.getLogger(FruitResource.class.getName());
+
+    @GET
+    public Uni<List<Fruit>> get() {
+        return Fruit.listAll(Sort.by("name"));
+    }
+
+    @GET
+    @Path("{id}")
+    public Uni<Fruit> getSingle(@RestPath Long id) {
+        return Fruit.findById(id);
+    }
+
+    @POST
+    public Uni<Response> create(Fruit fruit) {
+        if (fruit==null || fruit.id != null) {
+            throw new WebApplicationException("Id was invalidly set on request.", 422);
+        }
+
+        return fruit.persist()
+                .call(fruit::flush)
+                .onItem().transform(ignore -> Response.ok(fruit).status(201).build());
+    }
+
+    @PUT
+    @Path("{id}")
+    public Uni<Response> update(@RestPath Long id, Fruit fruit) {
+        if (fruit == null || fruit.getName() == null) {
+            throw new WebApplicationException("Fruit name was not set on request.", 422);
+        }
+
+        return Fruit.<Fruit>findById(id)
+                // If entity exists then
+                .onItem().ifNotNull()
+                    .transformToUni(entity -> {
+                        entity.setName(fruit.getName());
+                        return entity.flush().onItem().transform( ignore -> Response.ok( entity ).build() );
+                    } )
+                // else
+                .onItem().ifNull()
+                    .continueWith( Response.ok().status( 404 ).build() );
+    }
+
+    @DELETE
+    @Path("{id}")
+    public Uni<Response> delete(@RestPath Long id) {
+        return Fruit.findById(id)
```

Comment:
We have `deleteById`, would that be better?

Reply:
Is there a way to flush without having the entity?

Reply:
https://github.com/quarkusio/quarkus/issues/15595

Reply:
What I need here is something like `persistAndFlush`. It seems there is no  equivalent for the delete.
I could change the code to something like this:
```
    @DELETE
    @Path("{id}")
    public Uni<Response> delete(@RestPath Long id) {
        return Fruit.deleteAndFlushById(id)
                .onItem().transform( deleted -> deleted
                        ? Response.ok().status(204).build()
                        : Response.ok().status(404).build()
        );
    }
```

Reply:
Good point. I'll add it to the issue.

Reply:
> Hey, you're the DB expert, I defer to you ;)

:) Yeah, I think we need to show how to create transactions